v3/src/components/graph/models/axis-model.ts

```diff
@@ -1,22 +1,25 @@
 import {Instance, types} from "mobx-state-tree"
 import {GraphAttrPlace} from "./data-configuration-model"
 
-export const AxisPlaces = ["bottom", "left", "right", "top"] as const
+export const AxisPlaces = ["bottom", "left", "right", "top", "plot", "legend"] as const
```

Comment:
When we last had this conversation, I thought we had settled on the notion that `AxisPlaces` were places where axes could be and `GraphAttrPlaces` were places/roles that attributes could be. Now it seems like the ideas are getting muddled again.

For instance, `GraphModel` contains
```
    // keys are AxisPlaces
    axes: types.map(types.maybe(AxisModelUnion)),
```
so this change now implies that the graph has `plot` and `legend` axes. Maybe we need a new notion like
```
  export const GraphPlaces = [...AxisPlaces, "plot", "legend"]
```

Reply:
I followed your recommendation for GraphPlaces and also renamed PrimaryAttrPlaces to PrimaryAttrRoles, TipAttrPlaces to TipAttrRoles, GraphAttrPlaces to GraphAttrRoles, and GraphAttrPlace to GraphAttrRole